pkg/resource/rule/hooks_tags.go

```diff
@@ -0,0 +1,143 @@
+package rule
```

Comment:
Are these same tagging methods used by other resources with `EventBridge`? Would it be better to move them into a common package so that we aren't duplicating them?

Reply:
Yes, same for `EventBus` for example. Does the current code allow that without some form of generics? Do you have an example from another controller?

Reply:
`rds-controller` has a directory within `pkg/` for utility classes that they use across multiple resources (also related to tags) - https://github.com/aws-controllers-k8s/rds-controller/blob/main/pkg/util/tags.go If the types all line up, then you could use it like they do. You're also welcome to use generics now that we're standardised on 1.19, but I'm sure there are other ways around it.

Reply:
Yeah, looks like the tags for `Rule` and `EventBus` line up! Great suggestion, will create helper package.
jFYI: Generics won't help for now as they don't allow field accessors (in an easy way w/out introducing yet another interface which makes them kinda useless in such cases).